File: core/src/main/java/consulo/toml/ide/TomlHighlighterFactory.java
/*
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package consulo.toml.ide;

import consulo.annotation.component.ExtensionImpl;
import consulo.language.Language;
import consulo.language.editor.highlight.SyntaxHighlighter;
import consulo.language.editor.highlight.SyntaxHighlighterFactory;
import consulo.project.Project;
import consulo.virtualFileSystem.VirtualFile;
import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;
import org.toml.ide.TomlHighlighter;
import org.toml.lang.TomlLanguage;

@ExtensionImpl
public class TomlHighlighterFactory extends SyntaxHighlighterFactory {
    @Nonnull
    @Override
    public Language getLanguage() {
        return TomlLanguage.INSTANCE;
    }

    @Nonnull
    @Override
    public SyntaxHighlighter getSyntaxHighlighter(@Nullable Project project, @Nullable VirtualFile virtualFile) {
        return new TomlHighlighter();
    }
}

File: core/src/main/kotlin/consulo/toml/ide/TomlHighlighterFactory.kt
package consulo.toml.ide

import consulo.annotation.component.ExtensionImpl
import consulo.language.Language
import consulo.language.editor.highlight.SyntaxHighlighter
import consulo.language.editor.highlight.SyntaxHighlighterFactory
import consulo.project.Project
import consulo.virtualFileSystem.VirtualFile
import org.toml.ide.TomlHighlighter
import org.toml.lang.TomlLanguage

@ExtensionImpl
class TomlHighlighterFactory : SyntaxHighlighterFactory() {
    override fun getLanguage(): Language {
        return TomlLanguage
    }

    override fun getSyntaxHighlighter(p0: Project?, p1: VirtualFile?): SyntaxHighlighter {
        return TomlHighlighter();
    }
}